src/client/gui/lib/catalogue/launch_form.dart

```diff
@@ -245,11 +261,21 @@ class _LaunchFormState extends ConsumerState<LaunchForm> {
       ],
     );
 
+    final zones = ref.watch(zonesProvider);
+    final hasAvailableZones = zones.where((z) => z.available).isNotEmpty;
```

Comment:
Now that you don't need `zones` directly in this widget, you can inline the thing. Could you also move this at the top of the `build` method, where all the other `ref.watch`es are happening?
```suggestion
    final hasAvailableZones = ref.watch(zonesProvider).any((z) => z.available);
```